src/bin/startup-idps-metadata.ts

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env node
+
+import { array } from "fp-ts/lib/Array";
+import { fromNullable } from "fp-ts/lib/Option";
+import { task } from "fp-ts/lib/Task";
+import * as yargs from "yargs";
+import { logger } from "../utils/logger";
+import { fetchMetadataXML } from "../utils/metadata";
+
+//
+// parse command line
+//
+const argv = yargs
+  .option("idp-metadata-url-env", {
+    demandOption: false,
+    description: "ENV var name containing IDP Metadata URL",
+    normalize: true,
+    string: true
+  })
+  .option("testenv-metadata-url-env", {
+    demandOption: false,
+    description: "ENV var name containing TestEnv2 Metadata URL",
+    normalize: true,
+    string: true
+  })
+  .option("cie-metadata-url-env", {
+    demandOption: false,
+    description: "ENV var name containing CIE Metadata URL",
+    normalize: true,
+    string: true
+  })
+  .help().argv;
+
+interface IIDPSMetadataXML {
+  idps?: string;
+  xx_testenv2?: string;
+  xx_servizicie?: string;
+}
+
+function printIdpsMetadata(
+  idpsMetadataENV: string | undefined,
+  testEnv2MetadataENV: string | undefined,
+  cieMetadataENV: string | undefined
+): Promise<IIDPSMetadataXML> {
+  const maybeIdpsMetadataURL = fromNullable(idpsMetadataENV)
+    .mapNullable(_ => process.env[_])
+    .map((_: string) =>
+      fetchMetadataXML(_)
+        .map<{ idps?: string }>(_1 => ({
+          idps: _1.replace(/[\r\n\ ]+/g, "")
```

Comment:
it's not clear on why we replace all these spaces here. it's ok to setup multi-line env variables. moreover if we have string ie. "my organization name" I guess this will collapse this kind of statements.

Reply:
Good catch. I'll remove that code. I had added it only to get a more readable output.

Reply:
Fixed in 031a6f1